yarn-project/circuit-types/src/p2p/block_proposal.ts

```diff
@@ -0,0 +1,55 @@
+import { Header } from '@aztec/circuits.js';
+import { BaseHashType } from '@aztec/foundation/hash';
+import { BufferReader, serializeToBuffer } from '@aztec/foundation/serialize';
+
+import { TxHash } from '../index.js';
+import { Gossipable } from './gossipable.js';
+import { TopicType, createTopicString } from './topic_type.js';
+
+export class BlockProposalHash extends BaseHashType {
+  constructor(hash: Buffer) {
+    super(hash);
+  }
+}
+
+/**
+ * BlockProposal
+ *
+ * A block proposal is created by the leader of the chain proposing a sequence of transactions to
+ * be included in the head of the chain
+ */
+export class BlockProposal extends Gossipable {
+  static override p2pTopic: string;
+
+  constructor(
+    /** The block header, after execution of the below sequence of transactions */
+    public readonly header: Header,
+    /** The sequence of transactions in the block */
+    public readonly txs: TxHash[],
```

Comment:
I think this is good for now, but do we ultimately want to gossip the full TxObject? Since this may have come from a builder and the TxObject may not have been in the mempool?

Reply:
agreed, the granularity will likely need to change